app/src/main/java/com/github/wnder/MainActivity.java

```diff
@@ -29,6 +42,24 @@ protected void onCreate(Bundle savedInstanceState) {
 
         findViewById(R.id.getPictureButton).setOnClickListener(this);
         findViewById(R.id.uploadPictureButton).setOnClickListener(this);
+
+        String[] ss = {Manifest.permission.ACCESS_FINE_LOCATION};
+        ActivityCompat.requestPermissions(this, ss, 100); //Very important to have permission for future call
```

Comment:
Do we need this permission in the main activity or elsewhere like in the guess preview activity ?

Reply:
Having the permission at the beginning of the app allows us to use it at any time.